analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/InitializationVectorShouldBeRandomBase.cs

```diff
@@ -18,10 +18,108 @@
+        else
+        {
+            return state;
+        }
+    }
+
+    private static ProgramState ProcessAssignmentToIVProperty(IAssignmentOperationWrapper assignment, ProgramState state)
+    {
+        return assignment.Target?.AsPropertyReference() is { } property
+                && GetIVPropertyInstance(property) is { } instanceSymbol
+                && (assignment.Value.AsArrayCreation() is { } || state[assignment.Value].HasConstraint(ByteCollectionConstraint.CryptographicallyWeak))
+                    ? state.SetSymbolConstraint(instanceSymbol, ByteCollectionConstraint.CryptographicallyWeak)
+                    : null;
+
+        static ISymbol GetIVPropertyInstance(IPropertyReferenceOperationWrapper property) =>
+            property.Property.Name.Equals("IV")
+            && property.Instance.TrackedSymbol() is { } instanceSymbol
+            && instanceSymbol.GetSymbolType().DerivesFrom(KnownType.System_Security_Cryptography_SymmetricAlgorithm)
+               ? instanceSymbol
+               : null;
+    }
+
+    private static ProgramState ProcessStrongRandomGeneratorMethodInvocation(IInvocationOperationWrapper invocation, ProgramState state)
+    {
+        if (IsCryptographicallyStrongRandomNumberGenerator(invocation)
+            && invocation.ArgumentValue("data") is { } byteArray
+            && byteArray.TrackedSymbol() is { } byteArraySymbol)
+        {
+            return state.SetSymbolConstraint(byteArraySymbol, ByteCollectionConstraint.CryptographicallyStrong);
+        }
+        else if (IsGenerateIVMethod(invocation))
+        {
+            return state.SetSymbolConstraint(invocation.Instance.TrackedSymbol(), ByteCollectionConstraint.CryptographicallyStrong);
+        }
+        else
+        {
+            return null;
+        }
+    }
+
+    private ProgramState ProcessCreateEncryptorMethodInvocation(IInvocationOperationWrapper invocation, ProgramState state, SymbolicContext context)
+    {
+        if (IsCreateEncryptorMethod(invocation)
+            && (SecondArgumentIsCryptographicallyWeakArray(invocation, state) || HasCryptographicallyWeakIVProperty(invocation, state)))
+        {
+            ReportIssue(context.Operation.Instance, invocation.Instance.Syntax.ToString());
+        }
+        return state;
+
+        bool SecondArgumentIsCryptographicallyWeakArray(IInvocationOperationWrapper invocation, ProgramState state) =>
+            invocation.Arguments.Length == 2
+            && IvArgument(invocation) is { } ivArgument
+            && (ivArgument.AsArrayCreation() is { } || (ivArgument.TrackedSymbol() is { } symbol && state[symbol]?.HasConstraint(ByteCollectionConstraint.CryptographicallyWeak) is true));
+
+        bool HasCryptographicallyWeakIVProperty(IInvocationOperationWrapper invocation, ProgramState state) =>
+            invocation.Instance.TrackedSymbol() is { } instanceSymbol
+            && state[instanceSymbol]?.HasConstraint(ByteCollectionConstraint.CryptographicallyWeak) is true;
```

Comment:
Only check the property if no IV is passed as an argument! This reveals an FN that was hidden by an FP in the test cases.
Also, you can ditch the `TrackedSymbol()` here.

```suggestion
        bool UsesCryptographicallyWeakIVProperty(IInvocationOperationWrapper invocation, ProgramState state) =>
            invocation.Arguments.Length == 0
            && state[invocation.Instance]?.HasConstraint(ByteCollectionConstraint.CryptographicallyWeak) is true;
```

Reply:
But you can have this: 
```csharp
sa.IV = new byte[16];
sa.CreateEncryptor(sa.Key, sa.IV); // Noncompliant
```
You cannot raise here if  you check `invocation.Arguments.Length == 0`.
I will apply the other parts of the comment.

Reply:
As discussed offline thought - we need to actually cover this case separately so that we also cove
```csharp
sa.IV = new byte[16];
anotherObject.CreateEncryptor(sa.Key, sa.IV); // Noncompliant
```